Pi_Code/Controller_Operations.py

```diff
@@ -0,0 +1,113 @@
+        # We need to display a window in order to read joystick inputs.
+        pygame.joystick.init()
+        screen = pygame.display.set_mode((10, 10))
+
+        self.controller = pygame.joystick.Joystick(0)
+        self.controller.init()
+
+        self.idle = False # TODO: Change idle to true and add state switching in the main Submarine script
+        # Currently we'll just skip right past the idle stage and right into the auto sub stage. Idle is intended to be
+		# where we click the "start" button on the joystick to begin the operation of the sub
+        self.read_stick = True
+        self.autosub = True
+
+        # Get the properties of the joystick
+        # Not necessary currently, may be useful later
+        print("Found joystick: " + self.controller.get_name())
+        buttons = self.controller.get_numbuttons()
+        axes = self.controller.get_numaxes()
+        balls = self.controller.get_numballs()
+        hats = self.controller.get_numhats()
+        p = Process(target = self.getStickPosition, args = (dataFile, connection, SaveSensorData))
+        p.start()
+
+    def getStickPosition(self, connection, dataFile, SaveSensorData):
+
+        while self.idle:
+            # Start button code. Hold in idle until button 6 is pressed, and then send the response to the main loop
+            pygame.event.pump()
+            start = self.controller.get_button(6)
+            if start:
+                connection.send("auto")
+                self.idle = False
+
+        while self.autosub:
+            # If we press button 5, we will switch to manual controls.
+			# Otherwise, it will just keep looping in here.
+            pygame.event.pump()
+            manual = self.controller.get_button(5)
+            if manual:
+                connection.send("manual")
+                self.autosub = False
+                self.read_stick = True
+
+        while self.read_stick:
+            # Update states
+            pygame.event.pump()
+
+            # Hardcoded for now, will fix later.
+            threshhold = .3
+
+            roll = self.controller.get_axis(0)
+            pitch = self.controller.get_axis(1)
+            throttle = self.controller.get_axis(2)
+            # yaw = controller.get_axis(3) (Not used at the moment)
+            trigger = self.controller.get_button(0)
+            stop = self.controller.get_button(6)
+            kill = self.controller.get_button(1)
+            auto_revert = self.controller.get_button(5)
+
+            # Now that we have pitch/roll/throttle/yaw values, we can start using them
+            # Send a tuple (x,y) where x and y are the speeds for the left and right motors respectively
+
+            if kill:
+                # Shut down both the motors
+                connection.send((0, 0))
+
+            elif trigger:
+                # Set the speed to throttle and sent it to both motors
+                connection.send((throttle, throttle))
```

Comment:
Are you formatting the throttle so it's a value form 0-100 ? Negatives numbers and numbers that are not percentages won't work

Reply:
`throttle = self.controller.get_axis(2)` gives us some value between ~(-.7)-(+.54), not 0-1 or 0-100. I changed it so that it is multiplied by 100 so we should get between (-70) and (54). I take the absolute of the number when I send it in `sendLeft/RightSpeedPacket` and use the sign to figure out whether we go forward or backward.